### PR TITLE
Jk multinode

### PR DESCRIPTION
**Pull request by kuba-krj**
# Description
Add support for multinode training.
Neptune link: [Neptune](https://app.neptune.ai/o/pmtest/org/llm-random/runs/details?viewId=9a15aa36-1063-49c4-807d-53393abb118a&detailsTab=monitoring&shortId=LLMRANDOM-23471&type=run)

# Checklist
   - [x] Is this PR focused on a single feature without the possibility of being divided into smaller PRs?
   - [x] Are all variables set with logical defaults that are backward compatible?
   - [x] I attached link to neptune if it was necessary
   - [x] Have you successfully executed the linter and tests?
   - [x] Are all functions concise and don't require splitting?
   - [x] Is there documentation for arguments and complex logic?
   - [x] Are there no temporary solutions in the code? Should a task be added to Trello to refine the code?
   - [x] Are all functions placed in appropriate files?
   - [x] Is the PR name meaningful and descriptive?
   - [x] Is PR description provided, or unnecessary?
  
Other questions:
   - [x] If I made changes to the requirements or anything related to the Singularity image I did generate new image and upload a new image to the clusters
   - [ ] If this change is significant enough I notified all individuals working with this repository (@channel in slack channel)
   - [x] If there is a need for a second reviewer I requested additional review 

# Reviewer's checklist:
   - [ ] Does the Neptune link work and does it actually compare the situation before and after the PR?
   - [ ] Do I understand the code?
   - [ ] Do I think another reviewer is not needed or I commented that this PR need another reviewer?

### CODE REVIEW

File: lizrd/grid/grid_entrypoint.sh [hunk 1]
```diff
@@ -1,7 +1,22 @@
...
+    arg="${arg//__HEAD_NODE_IP__/$head_node_ip}"
+    arg="${arg//__RANDOM__/$RANDOM}"
+    args+=( "$arg" )
+done
+
+export LOGLEVEL=INFO
+
 module CUDA/11.7.0
 echo "Will run the following command:"
 echo "$@"
```
Comment (maciek-pioro) [inline lizrd/grid/grid_entrypoint.sh:20]: Should probably be `"${args[@]}"`

File: lizrd/train/checkpoints_manager.py [hunk 2]
```diff
@@ -188,6 +188,7 @@ def start_job_manager_assessment(
         else:
             return result, metadata
     else:
+        sleep(60)  # TODO: implement proper file locking
```
Comment (maciek-pioro) [inline lizrd/train/checkpoints_manager.py:191]: Is this intentional?

Comment (kuba-krj) [reply lizrd/train/checkpoints_manager.py:191]: Yes, it was advised by @madragonse

File: research/conditional/train/cc_train.py [hunk 3]
```diff
@@ -259,15 +263,12 @@ def convert_lr_scheduler_args(args, rampup_config):
 
 def main(
     rank: Optional[int],
```
[the remark this answers is not in the document]
Comment (kuba-krj) [reply research/conditional/train/cc_train.py:265]: Yes, it is needed when we do not launch the job using torchrun `mp.spawn` doesn't set any env varibles to let us know which rank we are on

**Review by maciek-pioro — APPROVED** [review]
Some small comments, but it should be fine
Very useful work, LGTM!

**Pull request merged**